src/cfnlint/schema_manager.py

```diff
@@ -0,0 +1,87 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+
+import json
+import logging
+import os
+
+import botocore.exceptions
+import boto3
+from cfnlint.template import Template
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchemaManager(object):
+    """Download schemas if necessary and validate modules"""
+    def __init__(
+            self, filename, template, regions):
+        self.filename = filename
+        self.regions = regions
+        self.template = Template(filename, template, regions)
+        self.boto3_sts = boto3.client('sts')
+        self.boto3_cfn = boto3.client('cloudformation')
+
+    # Check if the appropriate folder already exists
+    def check_folders(self, name, registry_type):
+        account_id = self.boto3_sts.get_caller_identity().get('Account')
+        for region in self.regions:
+            path = self.create_path(account_id, region, name)
+            if not os.path.isdir(path):
```

Comment:
So we only download the metadata once?  What if the version of the resource type changes?

Reply:
In a future PR, the --update-registry-type-specs parameter will allow the customer to refresh their locally cached registry type information. A check of the current version id will also throw a warning message to the customer to notify them that a later version is available in the CloudFormation registry every time that they validate their private registry types.